aicsimageio/transforms.py

```diff
@@ -105,59 +55,116 @@ def reshape_data(
         would have order "STCZYX"
 
     """
-    # Get operator
-    if isinstance(data, da.core.Array):
-        operator = da
-    else:
-        operator = np
+    # Check for parameter conflicts
+    for dim in Dimensions.DefaultOrderList:
+        # return_dims='TCZYX' and fixed dimensions 'C=1'
+        # Dimension is in kwargs
+        # Dimension is an integer
+        # Dimension is in return dimensions
+        if isinstance(kwargs.get(dim), int) and dim in return_dims:
+            raise ConflictingArgumentsError(
+                f"Argument return_dims={return_dims} and "
+                f"argument {dim}={kwargs.get(dim)} conflict. "
+                f"Check usage."
+            )
 
-    # Check for conflicts like return_dims='TCZYX' and fixed channels 'C=1'
-    for dim in return_dims:
-        if kwargs.get(dim) is not None:
+        # return_dims='CZYX' and iterable dimensions 'T=range(10)'
+        # Dimension is in kwargs
+        # Dimension is an iterable
+        # Dimension is not in return dimensions
+        if (
+            isinstance(kwargs.get(dim), (list, tuple, range, slice))
+            and dim not in return_dims
+        ):
             raise ConflictingArgumentsError(
                 f"Argument return_dims={return_dims} and "
```

Comment:
Proposed message: (roughly) "When selecting a Range of dim, the dim must also appear in return_dims"